src/clj/puppetlabs/services/jruby/jruby_puppet_core.clj

```diff
@@ -227,7 +227,8 @@
    ;; For an explanation of why we need a separate agent for the `flush-instance`,
    ;; see the comments in jruby-puppet-agents/send-flush-instance
    :flush-instance-agent  (jruby-agents/pool-agent agent-shutdown-fn)
-   :pool-state            (atom (jruby-internal/create-pool-from-config config))})
+   :pool-state            (atom (jruby-internal/create-pool-from-config config))
+   :shared-terminus-state (java.util.concurrent.ConcurrentHashMap.)})
```

Comment:
nitpick: would prefer an import on this class, rather than the fully qualified reference here, just for consistency